main/vstring.h

```diff
@@ -30,7 +30,7 @@
 #endif
 #ifdef VSTRING_PUTC_MACRO
 #define vStringPut(s,c) \
-	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : 0), \
+	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : (void) 0 ), \
 	((s)->buffer [(s)->length] = (c)), \
 	((c) == '\0' ? 0 : ((s)->buffer [++(s)->length] = '\0')))
 #endif
```

Comment:
You could also use an inline function and it would execute at the same speed and has the advantages of not evaluating arguments multiple times and producing more useful compiler errors when misused.